packages/fluxible/utils/callAction.js

```diff
@@ -14,38 +14,43 @@ require('setimmediate');
  * otherwise, Promise invocation.
  */
 function callAction (actionContext, action, payload, done) {
-    var executeActionPromise = new Promise(function (resolve, reject) {
-        setImmediate(function () {
-            try {
-                var syncResult = action(actionContext, payload, function (err, result) {
-                    if (err) {
-                        reject(err);
-                    } else {
-                        resolve(result);
-                    }
-                });
-                if (isPromise(syncResult)) {
-                    syncResult.then(resolve, reject);
-                } else if (action.length < 3) {
-                    resolve(syncResult);
+    // Use a promise to force the action to be called async
+    var executeActionPromise = Promise.resolve().then(function () {
+        // Assume the action returns a promise since it doesn't take a callback
+        if (action.length < 3) {
+            return action(actionContext, payload);
```

Comment:
The action can be sync or return a promise here. Also if the action throws, it will be properly caught.